src/IIIFPresentation/Repository.Tests/Paths/PathParserTests.cs

```diff
@@ -38,4 +39,62 @@ public void GetAssetIdFromNamedQueryCanvasId_Throws_IfInvalidFormat(string incor
         act.Should().Throw<FormatException>()
             .WithMessage($"Unable to extract AssetId from {incorrectId}");
     }
+
+    [Theory]
+    [InlineData("https://dlcs.example/1/canvases/someId")]
+    [InlineData("someId")]
```

Comment:
`[InlineData("https://dlcs.example/1/canvases/someId?foo=bar")]` fails this test, results in id of `someId?foo=bar`